### PR TITLE
Tila 2118 - GraphQL updated scalar typedefinition

### PR DESCRIPTION
**Pull request by wuahi**
- Updates graphql-codegen scalar types to limit the amount of 'any' typings
 - 'Fixes' ts-check errors in ui and admin-ui
 - 'Fixes' Carousel component

### CODE REVIEW

**Comment by wuahi** [comment]
Would be good for both @siren and @kieferbonk to test this locally that nothing breaks. It includes quite alot of changes with the typedefinition changes.

**Review by siren — APPROVED** [review]

File: ui/components/reservation/ReservationInfoCard.tsx [hunk 1]
```diff
@@ -112,12 +112,18 @@ const ReservationInfoCard = ({
...
-      ? getReservationUnitPrice(reservationUnit, begin)
+      ? getReservationUnitPrice(reservationUnit, new Date(begin))
       : getReservationPrice(reservation?.price, t("prices:priceFree"));
 
   const shouldDisaplyTaxPercentage: boolean =
     reservation?.state === "REQUIRES_HANDLING"
-      ? getReservationUnitPrice(reservationUnit, begin, 0, false, true) !== "0"
+      ? getReservationUnitPrice(
+          reservationUnit,
+          new Date(begin),
```
Comment (kieferbonk) [inline ui/components/reservation/ReservationInfoCard.tsx:122]: Need to check truthiness.

File: ui/components/reservation/ReservationInfoCard.tsx [hunk 2]
```diff
@@ -112,12 +112,18 @@ const ReservationInfoCard = ({
 
   const price: string =
     reservation?.state === "REQUIRES_HANDLING"
-      ? getReservationUnitPrice(reservationUnit, begin)
+      ? getReservationUnitPrice(reservationUnit, new Date(begin))
```
Comment (kieferbonk) [inline ui/components/reservation/ReservationInfoCard.tsx:115]: Need to check truthiness.

File: ui/pages/reservation-unit/[id].tsx [hunk 3]
```diff
@@ -605,8 +605,8 @@ const ReservationUnit = ({
       getSlotPropGetter(
         reservationUnit.openingHours?.openingTimes,
         activeApplicationRounds,
-        reservationUnit.reservationBegins,
-        reservationUnit.reservationEnds,
+        new Date(reservationUnit.reservationBegins),
+        new Date(reservationUnit.reservationEnds),
```
Comment (kieferbonk) [inline ui/pages/reservation-unit/[id].tsx:609]: Need to check truthiness.

File: ui/pages/reservation-unit/[id].tsx [hunk 4]
```diff
@@ -658,8 +658,8 @@ const ReservationUnit = ({
           [new Date(start), subMinutes(new Date(end), 1)],
           openingHours?.openingTimes,
           activeApplicationRounds,
-          reservationBegins,
-          reservationEnds,
+          new Date(reservationBegins),
+          new Date(reservationEnds),
```
Comment (kieferbonk) [inline ui/pages/reservation-unit/[id].tsx:662]: Need to check truthiness.

File: ui/components/calendar/ReservationCalendarControls.tsx [hunk 5]
```diff
@@ -346,8 +346,8 @@ const ReservationCalendarControls = <T extends Record<string, unknown>>({
           [startDate, subMinutes(endDate, 1)],
           reservationUnit.openingHours?.openingTimes,
           activeApplicationRounds,
-          reservationUnit.reservationBegins,
-          reservationUnit.reservationEnds,
+          new Date(reservationUnit.reservationBegins),
+          new Date(reservationUnit.reservationEnds),
```
Comment (kieferbonk) [inline ui/components/calendar/ReservationCalendarControls.tsx:350]: Need to check truthiness.

File: admin-ui/src/component/my-units/UnitReservations.tsx [hunk 6]
```diff
@@ -112,8 +112,8 @@ const merge = (
           .map((reservation) => ({
             title: reservation.name as string,
             event: reservation,
-            start: reservation.begin,
-            end: reservation.end,
+            start: new Date(reservation.begin),
+            end: new Date(reservation.end),
```
Comment (kieferbonk) [inline admin-ui/src/component/my-units/UnitReservations.tsx:116]: (Might) need to check truthiness.

**Review by kieferbonk — CHANGES_REQUESTED** [review]

File: ui/components/reservation/ReservationInfoCard.tsx (hunk 1 above)
Comment (wuahi) [reply ui/components/reservation/ReservationInfoCard.tsx:122]: Is it needed here? according to 

`const { begin, end, taxPercentageValue } = reservation || {};`

begin is always string? or do you mean is it a invalid date?

File: admin-ui/src/component/my-units/UnitReservations.tsx (hunk 6 above)
Comment (wuahi) [reply admin-ui/src/component/my-units/UnitReservations.tsx]: There should be no situation where this is empty

File: ui/modules/reservation.ts [hunk 7]
```diff
@@ -312,8 +312,8 @@ export const isReservationReservable = (
       [new Date(start), subMinutes(new Date(end), 1)],
       openingHours?.openingTimes,
       activeApplicationRounds,
-      reservationBegins,
-      reservationEnds,
+      new Date(reservationBegins),
+      new Date(reservationEnds),
```
Comment (kieferbonk) [inline ui/modules/reservation.ts:316]: I think you fixed this before the function was refactored, but these values also need truthiness check.

File: ui/components/reservation/EditStep0.tsx [hunk 8]
```diff
@@ -187,8 +187,8 @@ const EditStep0 = ({
       getSlotPropGetter(
         reservationUnit.openingHours?.openingTimes,
         activeApplicationRounds,
-        reservationUnit.reservationBegins,
-        reservationUnit.reservationEnds,
+        new Date(reservationUnit.reservationBegins),
+        new Date(reservationUnit.reservationEnds),
```
Comment (kieferbonk) [inline ui/components/reservation/EditStep0.tsx:191]: I think you fixed this before the function was refactored, but these values also need truthiness check.

**Review by kieferbonk — CHANGES_REQUESTED** [review]

**Pull request merged**